### ISSUE TITLE
Product Category can be recreated with the same name, description & color tag.

### ISSUE THREAD

**Issue by MehdiMsa** [idurar/idurar-erp-crm#1074]
**Describe the bug**
This problem is whenever a certain product category is created. Eventually the same product category could be recreated with same name, description & tag.

**To Reproduce**
1. Login to https://cloud.idurarapp.com/
2. Go to 'Products Category Section'
3. Click on 'Add New Product Category'
4. Fill in the information & create the Product Category
5. Click on 'Add New Product Category'
6. Recreate another Product Category with the same data

**Expected behavior**
A Message Suggesting that Product Category Already Exists

**Screenshots**
![image](https://github.com/idurar/idurar-erp-crm/assets/69842873/fb8e52d3-78d8-446b-a4ab-09edf89c0fb0)
![image](https://github.com/idurar/idurar-erp-crm/assets/69842873/6134d033-27bb-42fc-9fc5-16ad5d9bd27c)

**Desktop (please complete the following information):**
 - OS: Windows 11 Pro
 - Browser: Google Chrome
 - Version 

**Smartphone (please complete the following information):**
 - Version 123.0.6312.124 (Official Build) (64-bit)

**Comment by MehdiMsa** [comment]
Same goes for the Product Section, the Products could be duplicated too.
![image](https://github.com/idurar/idurar-erp-crm/assets/69842873/16b040c3-be74-4f5b-89f6-d0d7a6ca8bdd)

**Comment by yashashvi007** [comment]
Hey @MehdiMsa I replicated the issue, and I would like to work on it, can you please assign it to me.

**Comment by Houssemdaoud** [comment]
> Hey @MehdiMsa I replicated the issue, and I would like to work on it, can you please assign it to me.

i also replicated the issue and i would like to work on it if u haven't started @yashashvi007 or we could work on it together if you would like

**Comment by yashashvi007** [comment]
One question @MehdiMsa, on what parameters the uniqueness should be based on. Like should i consider name only because that will also work.

**Comment by Houssemdaoud** [comment]
> One question @MehdiMsa, on what parameters the uniqueness should be based on. Like should i consider name only because that will also work.

I think the uniqueness should be based on the color
Because we can have multiple food but different colors

**Comment by MehdiMsa** [comment]
@yashashvi007 @Houssemdaoud hello, feel free to take part of this work. I already started working on this issue 2 days ago & I would be happy to share the work with you.
And as @Houssemdaoud, the product category is classified through color selection as you mentions. By which I think could be considered as a parameter of uniqueness.

**Comment by MehdiMsa** [comment]
> Hey @MehdiMsa I replicated the issue, and I would like to work on it, can you please assign it to me.

Hey @yashashvi007, I do believe that only contributors with writers privilege can assign tasks for the issues. Nonetheless, feel free to take part on this task.

**Comment by yashashvi007** [comment]
Hey @MehdiMsa 

> @yashashvi007 @Houssemdaoud hello, feel free to take part of this work. I already started working on this issue 2 days ago & I would be happy to share the work with you. And as @Houssemdaoud, the product category is classified through color selection as you mentions. By which I think could be considered as a parameter of uniqueness.

Yes i would love to see your work on this one, I also done some work on it, I too love to share my work.

**Comment by marammahfoudhi** [comment]
hello @yashashvi007 @MehdiMsa is this issue worked on ?

**Comment by MehdiMsa** [comment]
@marammahfoudhi yes maram, I already fixed this issue. Nonetheless, I'm facing some issues pushing the code. Will be fixed by this week atmost.

**Comment by Ouchi-Ranim** [comment]
Hello Mhedi, I reproduced the issue and I'm facing the same problem, thanks for highlighting it.

**Comment by Ouchi-Ranim** [comment]
Did you push it yet?

**Comment by MehdiMsa** [comment]
> Hello Mehdi, I reproduced the issue and I'm facing the same problem, thanks for highlighting it.

Thank you for testing it out. I already pushed a fix for the solution in the following pull request:
[https://github.com/idurar/idurar-erp-crm/pull/1094/commits/d3b10928d0884116ad8d85ef9c25f57be2763073](Pull Request)

> Did you push it yet?
Yes indeed, you will find attached above the link to the changes made. Feel free to leave a code review. It will be very helpful.
Thank you & have a nice weekend.
